src/read.c

```diff
@@ -2760,10 +2723,6 @@ static void SaveReaderState(struct SavedReaderState *s) {
 }
 
 static void ClearReaderState(void ) {
-  ReaderState()->StackNams   = NEW_PLIST( T_PLIST, 16 );
-  ReaderState()->ReadTop     = 0;
-  ReaderState()->ReadTilde   = 0;
-  ReaderState()->CurrLHSGVar = 0;
```

Comment:
No need to preserve or restore these, as we stack allocate the `ReaderState` for each use freshly on the stack.